client/jetpack-cloud/sections/pricing/index.ts

```diff
@@ -1,15 +1,16 @@
 /**
  * Internal dependencies
  */
-import jetpackPlans from 'calypso/my-sites/plans/jetpack-plans';
 import * as controller from './controller';
+import jetpackPlans from 'calypso/my-sites/plans/jetpack-plans';
+import { siteSelection } from 'calypso/my-sites/controller';
 
 /**
  * Style dependencies
  */
 import './style.scss';
 
-export default function () {
-	jetpackPlans( `/:locale/pricing`, controller.jetpackPricingContext );
-	jetpackPlans( `/pricing`, controller.jetpackPricingContext );
+export default function (): void {
+	jetpackPlans( `/:locale/pricing/:site?`, siteSelection, controller.jetpackPricingContext );
+	jetpackPlans( `/pricing/:site?`, siteSelection, controller.jetpackPricingContext );
```

Comment:
The selected site id must be present in the app state in order for the cart to be properly synchronized. See https://github.com/Automattic/wp-calypso/blob/fb7c0fe1fdc3105009c88f7d8f9499f2f91e0032/client/lib/cart/store/index.js#L75.